deps/common/log/log.h

```diff
@@ -185,7 +189,9 @@ extern Log *g_log;
   if (common::g_log) {                                                     \
     struct timeval tv;                                                     \
     gettimeofday(&tv, NULL);                                               \
+    common::g_log->get_time_lock();                                        \
     struct tm *p                      = localtime(&tv.tv_sec);             \
```

Comment:
localtime is not thread-safe but there is a thread-safe interface called localtime_r. You can refer to manual book to check it.

Reply:
So just change localtime to localtime_r, no need for additional lock?

Reply:
That's the easiest method.
But localtime_r has bad performance, so it's welcomed if you have better solution.